File: Front-end/checko-frontend/src/models/MenuItem.ts
class MenuItem {
    id: number;
    name: string;
    description: string;
    price: number;
    image: string;
    calorie: number;
    category: string;
    lat: number;
    lng: number;
    count: number;
    secondHiCal: boolean;
  
    constructor(
      id: number,
      name: string,
      description: string,
      price: number,
      image: string,
      calorie: number,
      category: string,
      lat: number,
      lng: number,
      secondHiCal:boolean,
    ) {
      this.id = id;
      this.name = name;
      this.description = description;
      this.price = price;
      this.image = image;
      this.calorie = calorie;
      this.category = category;
      this.lat = lat;
      this.lng = lng;
      this.count = 0;
      this.secondHiCal = secondHiCal;
    }
  }
  
  export default MenuItem;
  

File: Backend/cheko/src/main/java/com/example/cheko/api/model/MenuItem.java
package com.example.cheko.api.model;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;

@Entity
@Table(name = "menu")

public class MenuItem {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;
    private String name;
    private String description;
    private double price;
    private String image;
    private int calorie;
    private String category;
    private double lat;
    private double lng;

    public MenuItem() {
    }

    public MenuItem(int id, String name, String description, double price, String image, int calorie, String category,
            double lat, double lng) {
        this.id = id;
        this.name = name;
        this.description = description;
        this.price = price;
        this.image = image;
        this.calorie = calorie;
        this.category = category;
        this.lat = lat;
        this.lng = lng;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public double getPrice() {
        return price;
    }

    public void setPrice(double price) {
        this.price = price;
    }

    public String getImage() {
        return image;
    }

    public void setImage(String image) {
        this.image = image;
    }

    public int getCalorie() {
        return calorie;
    }

    public void setCalorie(int calorie) {
        this.calorie = calorie;
    }

    public String getCategory() {
        return category;
    }

    public void setCategory(String category) {
        this.category = category;
    }

    public double getLat() {
        return lat;
    }

    public void setLat(double lat) {
        this.lat = lat;
    }

    public double getLng() {
        return lng;
    }

    public void setLng(double lng) {
        this.lng = lng;
    }

    @Override
    public String toString() {
        return "Menu{" +
                "id=" + id +
                ", name='" + name + '\'' +
                ", description='" + description + '\'' +
                ", price=" + price +
                ", image='" + image + '\'' +
                ", calorie=" + calorie +
                ", category='" + category + '\'' +
                ", lat=" + lat +
                ", lng=" + lng +
                '}';
    }
}
